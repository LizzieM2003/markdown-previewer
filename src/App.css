.App {
  width: 100vw;
  min-height: 100vh;
  box-sizing: border-box;
  display: flex;
  justify-content: space-around;
  padding: 5%;
  background-color: #80aaff; }
  .App #editor, .App #preview {
    width: 45%;
    padding: 2%;
    box-sizing: border-box;
    margin: 0; }
    @media (max-width: 800px) {
      .App #editor, .App #preview {
        width: 90%;
        height: 42vh; } }
  .App #preview {
    background-color: white;
    border: 1px solid #666; }
    @media (max-width: 800px) {
      .App #preview {
        overflow: scroll; } }
  @media (max-width: 800px) {
    .App {
      flex-direction: column;
      align-items: center;
      justify-content: space-between;
      min-height: 100vh; } }
