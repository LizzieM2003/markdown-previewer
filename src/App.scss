.App {
  width: 100vw;
  min-height: 100vh;
  box-sizing: border-box;
  display: flex;
  justify-content: space-around;
  padding: 5%;
  background-color: #80aaff;

  #editor, #preview {
    width: 45%;
    padding: 2%;
    box-sizing: border-box;
    margin: 0;

    @media (max-width: 800px) {
      width: 90%;
      height: 42vh;
    }
  }


  #preview {
    background-color: white;
    border: 1px solid #666;

    @media (max-width: 800px) {
      overflow: scroll;
      // height: 40vh;
    }
  }

  @media (max-width: 800px) {
    flex-direction: column;
    align-items: center;
    justify-content: space-between;
    min-height: 100vh;
  }
}




