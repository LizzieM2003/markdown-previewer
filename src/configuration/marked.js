import marked from 'marked';

marked.setOptions({
  breaks: true
});

export var renderer = new marked.Renderer();

renderer.link = function(href, title, text) {
  var out = '<a href="' + href + '"' + ' target="_blank"';
  if (title) {
    out += ' title="' + title + '"';
  }
  out += '>' + text + '</a>';
  return out;
};


export default marked;
