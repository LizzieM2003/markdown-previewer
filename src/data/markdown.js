const markdown =  `
Heading
=======

Sub-heading
---------

### Another deeper heading

Paragraphs are separated
by a blank line.

Leave 2 spaces at the end of a line to do a
line break  

Text attributes *italic*, **bold**,
\`monospace\`, ~~strikethrough~~.

> This is a blockquote.

Inline-style: 
![alt text](https://github.com/adam-p/markdown-here/raw/master/src/common/images/icon48.png "Logo Title Text 1")

Shopping list:

  * apples
  * oranges
  * pears

Numbered list:

  1. wash
  2. rinse
  3. repeat

The rain--not the reign--in
Spain.

 *[Herman Fasset](https://freecodecamp.com/hermanfassett)*
`;

export default markdown;