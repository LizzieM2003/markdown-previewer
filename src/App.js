import React, { Component } from 'react';
import marked, { renderer } from './configuration/marked';
import './App.css';
import markdown from './data/markdown';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      markdown
    };
  }

  render() {
    const markedHTML = { __html: marked(this.state.markdown, { renderer }) };

    return (
      <div className="App">
        <textarea
          id="editor"
          value={this.state.markdown}
          onChange={event => this.setState({ markdown: event.target.value })}
        />
        <div id="preview" dangerouslySetInnerHTML={markedHTML} />
      </div>
    );
  }
}

export default App;
