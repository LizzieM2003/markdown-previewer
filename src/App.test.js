import React from 'react';

import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import marked, { renderer } from './configuration/marked';

import App from './App';
import markdown from './data/markdown';

configure({ adapter: new Adapter() });

const app = shallow(<App />);

const markedHTML = marked(markdown, { renderer });
// console.log(markedHTML);

describe('App', () => {
  it('renders correctly', () => {
    expect(app).toMatchSnapshot();
  });

  it('has a textarea with id `editor`', () => {
    expect(app.find('textarea#editor').exists()).toBe(true);
  });

  it('has a preview element with id `preview`', () => {
    expect(app.find('#preview').exists()).toBe(true);
  });

  it('has an initial state set to markdown data', () => {
    expect(app.state().markdown).toBe(markdown);
  });

  it('sets the value of the textarea element to the markdown data', () => {
    expect(app.find('textarea#editor').props().value).toBe(markdown);
  });

  it('sets the HTML of the preview element to the converted markdown data', () => {
    expect(app.find('#preview').html()).toBe(
      '<div id="preview">' + markedHTML + '</div>'
    );
  });

  describe('when typing into the textarea', () => {
    const newMarkdown = `
      Heading
      =======
      
      ## Sub-heading
       
      Paragraphs are separated
      by a blank line.
      
      Two spaces at the end of a line  
      produces a line break.
      
      Text attributes _italic_, 
      **bold**, \`monospace\`.
      
      Horizontal rule:
      
      ---
      `;

    const newMarkedHTML = marked(newMarkdown);

    beforeEach(() => {
      app
        .find('textarea#editor')
        .simulate('change', { target: { value: newMarkdown } });
    });

    it('updates the `state`', () => {
      expect(app.state().markdown).toBe(newMarkdown);
    });

    it('updates the `preview` element with the changes', () => {
      expect(app.find('#preview').html()).toBe(
        '<div id="preview">' + newMarkedHTML + '</div>'
      );
    });
  });
});
