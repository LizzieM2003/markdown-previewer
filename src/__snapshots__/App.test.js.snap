// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <textarea
          id="editor"
          onChange={[Function]}
          value="
Heading
=======

Sub-heading
---------

### Another deeper heading

Paragraphs are separated
by a blank line.

Leave 2 spaces at the end of a line to do a
line break  

Text attributes *italic*, **bold**,
\`monospace\`, ~~strikethrough~~.

> This is a blockquote.

Inline-style: 
![alt text](https://github.com/adam-p/markdown-here/raw/master/src/common/images/icon48.png \\"Logo Title Text 1\\")

Shopping list:

  * apples
  * oranges
  * pears

Numbered list:

  1. wash
  2. rinse
  3. repeat

The rain--not the reign--in
Spain.

 *[Herman Fasset](https://freecodecamp.com/hermanfassett)*
"
/>,
        <div
          dangerouslySetInnerHTML={
                    Object {
                              "__html": "<h1 id=\\"heading\\">Heading</h1>
                    <h2 id=\\"sub-heading\\">Sub-heading</h2>
                    <h3 id=\\"another-deeper-heading\\">Another deeper heading</h3>
                    <p>Paragraphs are separated<br>by a blank line.</p>
                    <p>Leave 2 spaces at the end of a line to do a<br>line break  </p>
                    <p>Text attributes <em>italic</em>, <strong>bold</strong>,<br><code>monospace</code>, <del>strikethrough</del>.</p>
                    <blockquote>
                    <p>This is a blockquote.</p>
                    </blockquote>
                    <p>Inline-style:<br><img src=\\"https://github.com/adam-p/markdown-here/raw/master/src/common/images/icon48.png\\" alt=\\"alt text\\" title=\\"Logo Title Text 1\\"></p>
                    <p>Shopping list:</p>
                    <ul>
                    <li>apples</li>
                    <li>oranges</li>
                    <li>pears</li>
                    </ul>
                    <p>Numbered list:</p>
                    <ol>
                    <li>wash</li>
                    <li>rinse</li>
                    <li>repeat</li>
                    </ol>
                    <p>The rain--not the reign--in<br>Spain.</p>
                    <p> <em><a href=\\"https://freecodecamp.com/hermanfassett\\" target=\\"_blank\\">Herman Fasset</a></em></p>
                    ",
                            }
          }
          id="preview"
/>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "id": "editor",
          "onChange": [Function],
          "value": "
Heading
=======

Sub-heading
---------

### Another deeper heading

Paragraphs are separated
by a blank line.

Leave 2 spaces at the end of a line to do a
line break  

Text attributes *italic*, **bold**,
\`monospace\`, ~~strikethrough~~.

> This is a blockquote.

Inline-style: 
![alt text](https://github.com/adam-p/markdown-here/raw/master/src/common/images/icon48.png \\"Logo Title Text 1\\")

Shopping list:

  * apples
  * oranges
  * pears

Numbered list:

  1. wash
  2. rinse
  3. repeat

The rain--not the reign--in
Spain.

 *[Herman Fasset](https://freecodecamp.com/hermanfassett)*
",
        },
        "ref": null,
        "rendered": null,
        "type": "textarea",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "dangerouslySetInnerHTML": Object {
            "__html": "<h1 id=\\"heading\\">Heading</h1>
<h2 id=\\"sub-heading\\">Sub-heading</h2>
<h3 id=\\"another-deeper-heading\\">Another deeper heading</h3>
<p>Paragraphs are separated<br>by a blank line.</p>
<p>Leave 2 spaces at the end of a line to do a<br>line break  </p>
<p>Text attributes <em>italic</em>, <strong>bold</strong>,<br><code>monospace</code>, <del>strikethrough</del>.</p>
<blockquote>
<p>This is a blockquote.</p>
</blockquote>
<p>Inline-style:<br><img src=\\"https://github.com/adam-p/markdown-here/raw/master/src/common/images/icon48.png\\" alt=\\"alt text\\" title=\\"Logo Title Text 1\\"></p>
<p>Shopping list:</p>
<ul>
<li>apples</li>
<li>oranges</li>
<li>pears</li>
</ul>
<p>Numbered list:</p>
<ol>
<li>wash</li>
<li>rinse</li>
<li>repeat</li>
</ol>
<p>The rain--not the reign--in<br>Spain.</p>
<p> <em><a href=\\"https://freecodecamp.com/hermanfassett\\" target=\\"_blank\\">Herman Fasset</a></em></p>
",
          },
          "id": "preview",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <textarea
            id="editor"
            onChange={[Function]}
            value="
Heading
=======

Sub-heading
---------

### Another deeper heading

Paragraphs are separated
by a blank line.

Leave 2 spaces at the end of a line to do a
line break  

Text attributes *italic*, **bold**,
\`monospace\`, ~~strikethrough~~.

> This is a blockquote.

Inline-style: 
![alt text](https://github.com/adam-p/markdown-here/raw/master/src/common/images/icon48.png \\"Logo Title Text 1\\")

Shopping list:

  * apples
  * oranges
  * pears

Numbered list:

  1. wash
  2. rinse
  3. repeat

The rain--not the reign--in
Spain.

 *[Herman Fasset](https://freecodecamp.com/hermanfassett)*
"
/>,
          <div
            dangerouslySetInnerHTML={
                        Object {
                                    "__html": "<h1 id=\\"heading\\">Heading</h1>
                        <h2 id=\\"sub-heading\\">Sub-heading</h2>
                        <h3 id=\\"another-deeper-heading\\">Another deeper heading</h3>
                        <p>Paragraphs are separated<br>by a blank line.</p>
                        <p>Leave 2 spaces at the end of a line to do a<br>line break  </p>
                        <p>Text attributes <em>italic</em>, <strong>bold</strong>,<br><code>monospace</code>, <del>strikethrough</del>.</p>
                        <blockquote>
                        <p>This is a blockquote.</p>
                        </blockquote>
                        <p>Inline-style:<br><img src=\\"https://github.com/adam-p/markdown-here/raw/master/src/common/images/icon48.png\\" alt=\\"alt text\\" title=\\"Logo Title Text 1\\"></p>
                        <p>Shopping list:</p>
                        <ul>
                        <li>apples</li>
                        <li>oranges</li>
                        <li>pears</li>
                        </ul>
                        <p>Numbered list:</p>
                        <ol>
                        <li>wash</li>
                        <li>rinse</li>
                        <li>repeat</li>
                        </ol>
                        <p>The rain--not the reign--in<br>Spain.</p>
                        <p> <em><a href=\\"https://freecodecamp.com/hermanfassett\\" target=\\"_blank\\">Herman Fasset</a></em></p>
                        ",
                                  }
            }
            id="preview"
/>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "editor",
            "onChange": [Function],
            "value": "
Heading
=======

Sub-heading
---------

### Another deeper heading

Paragraphs are separated
by a blank line.

Leave 2 spaces at the end of a line to do a
line break  

Text attributes *italic*, **bold**,
\`monospace\`, ~~strikethrough~~.

> This is a blockquote.

Inline-style: 
![alt text](https://github.com/adam-p/markdown-here/raw/master/src/common/images/icon48.png \\"Logo Title Text 1\\")

Shopping list:

  * apples
  * oranges
  * pears

Numbered list:

  1. wash
  2. rinse
  3. repeat

The rain--not the reign--in
Spain.

 *[Herman Fasset](https://freecodecamp.com/hermanfassett)*
",
          },
          "ref": null,
          "rendered": null,
          "type": "textarea",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "dangerouslySetInnerHTML": Object {
              "__html": "<h1 id=\\"heading\\">Heading</h1>
<h2 id=\\"sub-heading\\">Sub-heading</h2>
<h3 id=\\"another-deeper-heading\\">Another deeper heading</h3>
<p>Paragraphs are separated<br>by a blank line.</p>
<p>Leave 2 spaces at the end of a line to do a<br>line break  </p>
<p>Text attributes <em>italic</em>, <strong>bold</strong>,<br><code>monospace</code>, <del>strikethrough</del>.</p>
<blockquote>
<p>This is a blockquote.</p>
</blockquote>
<p>Inline-style:<br><img src=\\"https://github.com/adam-p/markdown-here/raw/master/src/common/images/icon48.png\\" alt=\\"alt text\\" title=\\"Logo Title Text 1\\"></p>
<p>Shopping list:</p>
<ul>
<li>apples</li>
<li>oranges</li>
<li>pears</li>
</ul>
<p>Numbered list:</p>
<ol>
<li>wash</li>
<li>rinse</li>
<li>repeat</li>
</ol>
<p>The rain--not the reign--in<br>Spain.</p>
<p> <em><a href=\\"https://freecodecamp.com/hermanfassett\\" target=\\"_blank\\">Herman Fasset</a></em></p>
",
            },
            "id": "preview",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
